Automattic/wp-calypso
# Update module exports to use ES6 syntax

This pull request seeks to update all module exports that follow the [NPM syntax](https://nodejs.org/api/modules.html#modules_module_exports) to use the [ES6 syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export#Using_the_default_export). This basically means updating the following exports:

```javascript
module.exports = { ... };
module.exports = React.createClass( ... );
module.exports = Buttons;
```

Into :

```javascript
export default { ... };
export default React.createClass( ... );
export default Buttons;
```
  
#### Testing instructions
 
1. Run `git checkout update/client-exports-to-es6` and start your server, or open a [live branch](https://calypso.live/?branch=update/client-exports-to-es6)
2. Open the [`Home` page](http://calypso.localhost:3000/)
3. Check that everything works as before
 
#### Additional notes

I'd love to say that I spent my weekend updating the +2800 exports that are still using the old syntax manually. Unfortunately the reality is a bit different since it actually took me five minutes (once I knew what codemods are) to do that with:

```
npm install jscodeshift

node_modules/jscodeshift/bin/jscodeshift.sh client --extensions js,jsx --transform https://raw.githubusercontent.com/5to6/5to6-codemod/master/transforms/exports.js
```

#### Reviews
